src/main/java/uk/gov/hmcts/reform/em/stitching/rest/DocumentTaskResource.java

```diff
@@ -58,21 +62,49 @@ public DocumentTaskResource(DocumentTaskService documentTaskService) {
     public ResponseEntity<DocumentTaskDTO> createDocumentTask(
             @Valid @RequestBody DocumentTaskDTO documentTaskDTO,
             @RequestHeader(value = "Authorization", required = false) String authorisationHeader,
-            HttpServletRequest request) throws URISyntaxException {
+            HttpServletRequest request) throws URISyntaxException, IOException {
 
         log.info("REST request to save DocumentTask : {}, with headers {}", documentTaskDTO.toString(),
                 Arrays.toString(Collections.toArray(request.getHeaderNames(), new String[]{})));
 
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.registerModule(new JavaTimeModule());
+
+        String docTask = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(documentTaskDTO);
+        mapper.readValue(docTask, DocumentTaskDTO.class);
+        log.info("Received request body as Json \n" + docTask);
+
         if (documentTaskDTO.getId() != null) {
             throw new BadRequestAlertException("A new documentTask cannot already have an ID", ENTITY_NAME, "id exists");
         }
-        documentTaskDTO.setJwt(authorisationHeader);
-        documentTaskDTO.setTaskState(TaskState.NEW);
-        DocumentTaskDTO result = documentTaskService.save(documentTaskDTO);
+        DocumentTaskDTO result = null;
+        try {
+            documentTaskDTO.setJwt(authorisationHeader);
+            documentTaskDTO.setTaskState(TaskState.NEW);
+            result = documentTaskService.save(documentTaskDTO);
+
+        } catch (DataIntegrityViolationException e) {
+
+            log.error("Error while mapping entites for DocumentTask : " + e.getRootCause()
+                    + " DocumentTask contains " + docTask);
+            e.printStackTrace();
 
+            return
+                    ResponseEntity.badRequest()
+                            .body(result);
+        } catch (Exception ex) {
+            log.error("Error while mapping entites for DocumentTask : " + ex.getCause()
+                    + " DocumentTask contains :" + documentTaskDTO.toString() + docTask);
+            ex.printStackTrace();
+            return
+                    ResponseEntity.badRequest()
```

Comment:
can't you return this in the try?

Reply:
well instead of a log.info i wanted to send it alongwith the error , thats why kept + " DocumentTask contains :" + documentTaskDTO.toString() + docTask); in the relevant catch blocks ..

plus the return statements i have modified as below -  try {
            documentTaskDTO.setJwt(authorisationHeader);
            documentTaskDTO.setTaskState(TaskState.NEW);
            result = documentTaskService.save(documentTaskDTO);

            return ResponseEntity.created(new URI("/api/document-tasks/" + result.getId()))
                    .headers(HeaderUtil.createEntityCreationAlert(ENTITY_NAME, result.getId().toString()))
                    .body(result);

        } catch (DataIntegrityViolationException e) {

            log.error("Error while mapping entites for DocumentTask : " + e.getRootCause()
                    + " DocumentTask contains " + docTask);
            e.printStackTrace();
            return ResponseEntity.badRequest().body(null);

        } catch (Exception e) {
            log.error("Error while mapping entites for DocumentTask : " + e.getCause()
                    + " DocumentTask contains :" + documentTaskDTO.toString() + docTask);
            e.printStackTrace();
            return ResponseEntity.badRequest().body(null);
        }